process/smartContract/scQueryService.go

```diff
@@ -155,9 +156,16 @@ func (service *SCQueryService) ComputeScCallGasLimit(tx *transaction.Transaction
 		return 0, err
 	}
 
-	gasConsumed := service.economicsFee.MaxGasLimitPerBlock(0) - vmOutput.GasRemaining
+	if vmOutput.ReturnCode != vmcommon.Ok {
+		return 0, errors.New(vmOutput.ReturnMessage)
+	}
+
+	moveBalanceGasLimit := service.economicsFee.ComputeGasLimit(tx)
+	gasConsumedExecution := math.MaxUint64 - vmOutput.GasRemaining
```

Comment:
i do not like this hardcoded value. Why not using tx.GasLimit ? What happens if we decide that we call the SC query with a value different of maxUint?

Reply:
refactored this part.